ring/consul_client.go

```diff
@@ -206,6 +212,65 @@ func (c *consulClient) WatchPrefix(prefix string, factory func() interface{}, do
 	}
 }
 
+// WatchKey will watch a given key in consul for changes. When the value
+// under said key changes, the f callback is called with the deserialised
+// value. To construct the deserialised value, a factory function should be
+// supplied which generates an empty struct for WatchPrefix to deserialise
+// into. Values in Consul are assumed to be JSON. This function blocks until
+// the done channel is closed.
```

Comment:
ISTR you saying something to me about how it's not good Go practice to pass around channels. Am I misremembering?